mozci/data/contract.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+
+from dataclasses import dataclass
+from typing import Tuple
+
+from voluptuous import Any, Marker, Optional, Required, Schema
+
+
+@dataclass
+class Contract:
+    name: str
+    description: str
+    validate_in: Schema
+    validate_out: Schema
+
+
+_contracts: Tuple[Contract, ...] = (
+    Contract(
+        name="push_tasks",
+        description="Data about the tasks that ran on a given push.",
+        validate_in=Schema({Required("branch"): str, Required("rev"): str,}),
+        validate_out=Schema(
+            [
+                {
+                    Required("id"): str,
+                    Required("label"): str,
+                    Required("result"): str,
+                    Required("classification"): Any(
+                        "not classified", "intermittent", "fixed by commit",
+                    ),
+                    Optional("classification_note"): str,
+                    Optional("duration"): int,
+                }
+            ]
+        ),
+    ),
+    Contract(
+        name="push_test_groups",
+        description="Data about the test groups that ran on a given push.",
+        validate_in=Schema({Required("branch"): str, Required("rev"): str,}),
+        validate_out=Schema(
```

Comment:
This will likely involve Treeherder downloading the artifact.  Though @ahal had mentioned that the same will likely be true for some other datasources.  So perhaps we won't need to implement this on the treeherder side.